src/job.rs

```diff
@@ -55,54 +60,42 @@ impl Job {
     /// Return basic and extended limit information for a job object.
     /// See also [Microsoft Docs](https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-jobobject_extended_limit_information).
     pub fn query_extended_limit_info(&self) -> Result<ExtendedLimitInfo, JobError> {
-        let mut info = ExtendedLimitInfo::new();
+        let mut info = ExtendedLimitInfo::default();
 
-        let return_value = unsafe {
+        unsafe {
             QueryInformationJobObject(
                 self.handle,
                 JobObjectExtendedLimitInformation,
-                &mut info.0 as *mut _ as LPVOID,
-                mem::size_of_val(&info) as DWORD,
-                0 as *mut _,
+                &mut info.0 as *mut _ as *mut c_void,
+                mem::size_of_val(&info.0) as u32,
+                None,
             )
-        };
-
-        if return_value == 0 {
-            Err(JobError::GetInfoFailed(io::Error::last_os_error()))
-        } else {
-            Ok(info)
+            .map_err(|e| JobError::GetInfoFailed(e.into()))?;
```

Comment:
Small nit: can you move the `map_err`s out of the `unsafe` blocks?

Reply:
Done